src/tribler-core/tribler_core/modules/metadata_store/store.py

```diff
@@ -149,6 +149,17 @@ def sqlite_disable_sync(_, connection):
                 default_vsids = self.Vsids.create_default_vsids()
             self.ChannelMetadata.votes_scaling = default_vsids.max_val
 
+        with db_session:
+            self.pk_infohash_uniq_index_exists = False
+            for ind in range(0, 10):
```

Comment:
Why 10?

Reply:
just random number that is more than the possible number of indexes that could have been created by SQLite automatically?

Reply:
Ok. Is there any simple way to get the number of indices from sqlite? If not, at the very least please add a comment that this value should be higher than the number of existing indices 👍

Reply:
> Is there any simple way to get the number of indices from sqlite?

If you just want to list the indices you can run this:

```sql
SELECT name FROM sqlite_master WHERE type='index'
```

Reply:
I knew that at least *one* of the team will suggest this :facepalm:  Professional programmers' response :fist_raised: 

Yes, querying the number of indices will limit the number of checks to *about 20* checks, that is far *more* than 10. So it will not optimize anything (mind you, it is hard to imagine that the thing must be optimized). I doubt there is, or was, or ever will be (except for Quinten-factor, but where would we be if we always think about that?) any Tribler installation of 7.5.x featuring more than 10 autoindexes.

So, the only option to solve this properly is to add a query that will count the number of autoindexes in the list.
Personally, I am almost ashamed to spend even *these* letters, yours and my time to discuss and *solve* such a trivial "problem".

But if you guys insist, I will solve this for good :shrug:   :wink:

Reply:
> except for Quinten-factor

Lol 😄, even I don't try to bend over backwards to get sqlite to make extra indices (I only have 3 as you can see). It is indeed quite unlikely anyone would have more than 10 autoindices.

That said, I agree with @devos50: if 2~3 easy extra lines of code makes this code work 100% of the time instead of 99% of the time, I'd vote to spend an extra 2 minutes on this PR and solve it for good.

Reply:
> Personally, I am almost ashamed to spend even these letters, yours and my time to discuss and "solve" such a trivial "problem".

Isn't that the whole point of PR reviews? When I review a PR, I point out all issues that can arise, both major and minor. In this situation, the issue is very minor but it is an issue nonetheless and something that can trivially be fixed. If you have a different opinion on how we should treat PR reviews, please let me know and I can take it into consideration.

You indicated that you already knew that one of us would identify this issue, which (in my opinion) should be reason enough to have addressed this "trivial" issue before opening a PR.

Reply:
> If you have a different opinion on how we should treat PR reviews, please let me know and I can take it into consideration.

Whenever I review a PR, when I notice a risk or potential problem, or another kind of opportunity for improvement, before even *notifying* the author about the thing, I consider the tradeoff between:
 1. the potential impact of the change (e.g. how big it the risk that )
 2. the amount of energy the author and the team as a whole will have to spend on fixing that.

That's called [leverage](https://gist.github.com/rondy/af1dee1d28c02e9a225ae55da2674a6f). It is the most important concept in the Effective Engineer book.

If the tradeoff does not look good, I don't notify the author. Thus, I conserve their time, mental energy, and provide as little obstructions to their thinking process as possible. That's why I, e.g. never correct grammar mistakes in other's PRs: *it is just not worth it*. 



Changing even a single letter requires much more than changing a letter in the text editor these days. It requires:
1. committing the change
2. squashing the changes
3. requesting another re-review
4. rebasing the PR (because while you were doing steps 1-3 someone else already committed their stuff before you)
5. requesting another re-review because of the previous step

Pipelining multiple PRs does not help either, because it forces one to multitask, which is very inefficient and taxing on mental energy, and you guarantee to force step 4 on yourself (rebasing your work on top of you own work that was just accepted).

So, that's my position on both writing *and* reviewing code: maximize *leverage*, skip stuff with low *leverage* completely.